_posts/aws/2019-06-03-AWS-api-gateway-integration.md

```diff
@@ -0,0 +1,83 @@
+
+In this case we will expose a webhook URL to trigger a pipeline which could be called from a third-party system outside of our infrastructure, this document also assumes that you have configured Spinnaker in EKS.
+
+## Adding a webhook trigger in Spinnaker
+
+First, we need to add a trigger to our pipeline, under Configuration, select Add Trigger and make its type selector Webhook, this will cause Spinnaker to expose an endpoint that must be hit in order to trigger the pipeline, for more information about configuring a webhook trigger check [here](https://www.spinnaker.io/guides/user/pipeline/triggers/webhooks/) .
+
+<a href="https://cl.ly/ccd8e0a71a61" target="_blank"><img src="https://d2ddoduugvun08.cloudfront.net/items/0L0k2G382g1H092V3f2W/Screen%20Shot%202019-06-03%20at%203.52.56%20PM.png" style="display: block;height: auto;width: 100%;"/></a>
+
+## Creating AWS API Gateway
+
+Next, we can proceed in the creation of an API in AWS API Gateway, the main intention of this is to maintain security and control over Spinnaker's exposed endpoints, and API Gateway offers an easy way to create, publish, maintain, monitor, and secure APIs at any scale.
+
+<a href="https://cl.ly/e9d0ae7fb024" target="_blank"><img src="https://d2ddoduugvun08.cloudfront.net/items/1X3z1w2A392o3l0L2C2U/Screen%20Shot%202019-06-03%20at%203.17.58%20PM.png" style="display: block;height: auto;width: 100%;margin-bottom:10px"/></a>
+
+You might notice after have deployed your API, that establishing a connection between AWS API Gateway to Spinnaker it’s not straight forward, this is mainly because API Gateway runs in its own VPC and is completely managed by AWS, removing the option of VPC peering between both networks. 
+
+## Creating Network Load Balancer
+
+Fortunately, with AWS API Gateway, you can do [“Private Integrations”](https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-private-integration.html), which is a way to expose your HTTP/HTTPS resources behind an Amazon VPC for access by clients or third party systems outside of the VPC, currently only Network Load Balancers are supported to make private integrations.
+
+<a href="https://cl.ly/02e7d06c8a7a" target="_blank"><img src="https://d2ddoduugvun08.cloudfront.net/items/3D082j3U1u431F1k2L1y/Screen%20Shot%202019-06-03%20at%2011.15.54%20PM.png" style="display: block;height: auto;width: 75%;"/></a>
+
+Before create a Network load balancer, Let's create a NodePort that opens a specific port on all worker nodes, and any traffic that is sent to this port will be forwarded to the Gate's service.
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: spin-gate-internal
+  namespace: default
```

Comment:
You can put a placeholder here like `namespace: <spinnaker's namespace>`, since it will not always be `default`